### ISSUE TITLE
go-restful upgrade incompatible for apiserver

### ISSUE THREAD

**Issue by Icarus9913** [kubernetes/kubernetes#118133]
### What happened?

Once I build `sample-apiserver` project with go module updrage(`go-restful` from v3.9.0 to v3.10.0) in my cluster, it doesn't work well.

```text
NAME                                   SERVICE      AVAILABLE                      AGE
v1alpha1.wardle.example.com            wardle/api   False (FailedDiscoveryCheck)   9s
---
$ kubectl describe apiservice v1alpha1.wardle.example.com
···
Spec:
  Group:                     wardle.example.com
  Group Priority Minimum:    1000
  Insecure Skip TLS Verify:  true
  Service:
    Name:            api
    Namespace:       wardle
    Port:            443
  Version:           v1alpha1
  Version Priority:  15
Status:
  Conditions:
    Last Transition Time:  2023-05-19T09:36:54Z
    Message:               failing or missing response from [https://172.40.1.34:443/apis/wardle.example.com/v1alpha1](https://172.40.1.34/apis/wardle.example.com/v1alpha1): bad status from [https://172.40.1.34:443/apis/wardle.example.com/v1alpha1](https://172.40.1.34/apis/wardle.example.com/v1alpha1): 404
    Reason:                FailedDiscoveryCheck
    Status:                False
    Type:                  Available
Events:                    <none>
···
```

### What did you expect to happen?

work good even the dependency updated

### How can we reproduce it (as minimally and precisely as possible)?

just update the go module `go-restful` from v3.9.0 to v3.10.0.

### Anything else we need to know?

`go-restful` v3.10.0 git commit: https://github.com/emicklei/go-restful/commits/v3.10.0

### Kubernetes version

<details>

```console
$ kubectl version
main branch
```

</details>

### Cloud provider

<details>

</details>

### OS version

ubuntu

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by Icarus9913** [comment]
cc @liggitt @sttts 
/triage accepted

**Comment by liggitt** [comment]
/remove-kind bug
/kind cleanup

this seems more like a compatibility bug in that library to me.

**Comment by liggitt** [comment]
see attempt to bump in https://github.com/kubernetes/kubernetes/pull/115067

verify failed because path join behavior changed

it looks like 3.10.2 intentionally leaves behavior incompatible with 3.9 and lets callers opt back into compatible behavior
* https://github.com/emicklei/go-restful/blob/v3/CHANGES.md#v3102---2023-03-09
* https://github.com/emicklei/go-restful/blob/v3/README.md#how-to-customize

**Comment by Icarus9913** [comment]
But I just upgrade `go-restful` from v3.9.0 to v3.10.0
Why v3.10.0 starts to break it, a little confused.

**Comment by liggitt** [comment]
because go-restful made a breaking change in 3.10.0

**Comment by Icarus9913** [comment]
oh, bad.
I didn't notice the title `[v3.10.0] - 2022-10-11 - BROKEN`
I just checked its commit codes.

**Comment by Icarus9913** [comment]
Reference: https://github.com/emicklei/go-restful/issues/497

**Comment by amaciejk** [comment]
Any chance this will be resolved as we need go-restful to bump to v3.10 or higher to resolve PRISMA-2022-0227?

**Comment by Icarus9913** [comment]
If I remember correctly, there's a patch version of `go-restful` resolved this incompatible.
